### DISCUSSION TITLE
INSTAGRAM/FACEBOOK GRAPH API - Is there a Webhook available to watch for 'new posts' on Instagram?

### DISCUSSION THREAD

**Discussion by glockindacroc611** [community/community#113223] [Programming Help]
Currently in the developing stages of a lowcode web app that will allow users to connect their Instagram & Facebook via OAuth then enable a backend automation (with third party app) that will send new IG posts to their Facebook. 

Intended to be a cross-posting app. 

Is there an IG Graph Webhook available for this? Or will I need to set up an API call from my app to watch for the new posts.
I also have an app created & verified with Meta already as a business app. I'm wondering if this was the right decision to create it as a business or should it be a consumer app.

When subscribing to a webhook, I was not able to locate anything to subscribe to 'new posts'.
If the next option is to set up an API call to watch for new posts for every 'x' amount of minutes is there an endpoint and headers that is provided in the docs to set this up?

Any other avenues on how to successfully get this feature up would be greatly appreciated.

Also want to note, I'm a beginner at programming so I'm going at this with very little prior knowledge. Mostly have been researching through developer docs & have not gotten any success. I have been meaning to launch this app for the past month, but have been getting stuck in the programming phase with the front end ready for the most part.

TY in advance!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by lin20260407** [comment]
hey bro 

To automate cross-posting from Instagram to Facebook, you can utilize the Instagram Graph API. Currently, there isn't a direct Webhook subscription for "new posts" notifications on Instagram, but you can indirectly monitor updates by subscribing to changes in the "media" object, which might include new post creations.

For precise tracking of new posts, a polling approach via the API is recommended. Regularly request the user's media list from Instagram and check for posts added since your last check.

Registering your app as a business app is the correct choice if it primarily serves Instagram Business or Creator accounts and requires access to specific Instagram Graph API features for these account types.

Steps for implementing polling include:

Regular Requests: Periodically make requests to the /{ig-user-id}/media endpoint to fetch the user's media list.
Compare Timestamps: Save the timestamp of your last check and compare it with the timestamps of the retrieved media objects to identify new posts.
Take Action: For each newly discovered post, perform the desired action, such as posting it to Facebook.

**Reply by glockindacroc611** [reply]
hey bro thanks so much for your response! it's been a struggle getting help from other communities. 

going to implement everything you suggested and will let you know how everything goes! 

hoping to get my app launched before the end of next week.

**Reply by An88ton** [reply]
Hi @weitsunglin, I have the same problem can you say more detail about subscribing to media object changes, how can I do it, because I did not find any resources about it, and in webhook configurations there is no such field seems, I have business type app.

**Comment by IsuruSankhajith** [comment]
To enable cross-posting functionality between Instagram and Facebook within your lowcode web app, several technical considerations need to be addressed. Firstly, regarding Instagram Graph API's webhook availability, it's essential to note that while the API provides webhooks for certain events, such as user activity, there isn't a specific webhook for new posts. Therefore, implementing a solution may involve setting up an API call from your app to periodically check for new posts. This approach entails creating a scheduled task within your application to make requests to the Instagram Graph API at regular intervals, querying for new posts since the last check.

Given that you've already created and verified your app with Meta as a business app, it's important to evaluate whether this classification aligns with your app's intended use case and audience. Business apps typically cater to commercial or organizational purposes, whereas consumer apps target individual users. Consider whether the features and functionalities of your app better suit the business or consumer category, as this classification may affect how your app interacts with Instagram and Facebook APIs.

When implementing the API call to watch for new posts, refer to the official documentation provided by Instagram for detailed instructions on endpoint URLs, request parameters, and required headers. These resources will guide you in constructing the appropriate API requests to retrieve the desired data efficiently. Additionally, leverage community forums, developer communities, and online tutorials to gain insights from experienced developers and troubleshoot any challenges you encounter along the way.

As a beginner in programming, navigating through developer documentation and troubleshooting technical issues can be daunting. Consider seeking assistance from mentors, joining online forums, or participating in coding communities where you can receive guidance and support from experienced developers. Breaking down tasks into smaller, manageable steps and gradually building upon your knowledge will help you progress towards launching your app successfully. Keep iterating on your codebase, testing functionality, and seeking feedback to refine your app and achieve your development goals. With dedication and perseverance, you'll overcome programming hurdles and bring your cross-posting feature to fruition.

**Comment by aimanoorai** [comment]
Instagram does not officially provide a webhook specifically for monitoring "new posts." However, apps like[ Honista APK](https://honistahub.net/) might use unofficial APIs or techniques to monitor Instagram activities, which may violate Instagram's terms of service.

**Comment by hubas900** [comment]
You're right—Instagram’s Graph API doesn’t offer a webhook for “new posts” from user profiles. Most developers set up scheduled API calls to /me/media to check for new content every few minutes. Just be mindful of rate limits. As for app type, Business is the right choice if you're automating features for Instagram Professional accounts.

Apps like [Honista Insta latest version](https://honistaap.pro/) don’t rely on the official API—they tap directly into Instagram’s interface for features like post downloads or anonymous viewing, but that comes with security tradeoffs. If you're aiming to stay within Meta's terms, API polling with proper tokens and rate management is the way to go. Let me know if you want a sample call structure.

**Comment by midhun98** [comment]
yes seems like there is no official way for this. From what I have gone through the api documentation we should be using this api endpoint
https://graph.instagram.com/v23.0/me/media?access_token={{long_access_token}}&fields=your_fields

to keep uptodate with latest posts. Also these are the only webhooks events provided by meta
https://developers.facebook.com/docs/graph-api/webhooks/reference/instagram

**Comment by hubas900** [comment]
Yes, Instagram’s Graph API supports webhooks for specific events like new media posts, but it’s limited to Business and Creator accounts. For [entrepreneurs using platforms](https://www.meetworth.com/) like Meetworth to grow their brand or startup, integrating such tools can help monitor content engagement and stay active in networking through social media.

**Comment by aristotal34** [comment]
Hey! Since you’re working on managing Instagram content and cross-posting, you might find some helpful tools at [visit site](https://honista.social/). It’s a fully customized Instagram app that allows you to save reels, posts, and stories easily. You can also view profiles anonymously, see full-size display pictures, and enjoy additional privacy features. For beginners experimenting with Instagram features, exploring such apps can give useful ideas and help you understand how content management works more effectively.

Additionally, Honista offers customization options like text colors, fonts, and custom emojis, along with iOS-inspired features such as rounded-edge stories. While it doesn’t replace API functionality for automation, it’s a great resource to learn from and get inspiration for your app. It’s safe for both your device and Instagram account.

**Comment by div-dhingra** [comment]
@glockindacroc611 Hey man, any updates? did they add webhook support or still polling? Were you able to access the Meta API Key past review? THANKS!
